### ISSUE TITLE
projection parameters from grib file

### ISSUE THREAD

**Issue by nikhil003** [jswhit/pygrib#167]
I am trying to process some grib files, where I need to get the projection information from the file. When doing so, I extract the projparams for the grib message, which is a dictionary with projection parameters. Afterwards, I try to instantiate a Proj object but get error when try to do so.

```python
import pygrib
import pyproj
fin = pygrib.open('./IDY25026.APS3.wind.10m.2019092412.006.surface.grb2')
grbs = [grb for grb in fin]
pyproj.Proj(grbs[0].projparams)
```

I have tried this with pyproj version 1.9.6 and pygrib version 2.0.3

>     Traceback (most recent call last):
>     File "", line 1, in
>     File "/home/gar305/Documents/geostack/pyinstall/lib/python3.7/site-packages/pyproj/init.py", line 362, in new
>     return _proj.Proj.new(self, projstring)
>     File "_proj.pyx", line 129, in _proj.Proj.cinit
>     RuntimeError: b'unknown projection id'
> 

The projparams are

>     {'a': 6371229, 'b': 6371229, 'proj': 'cyl'}

The files I am testing with is ftp://ftp.bom.gov.au/register/sample/access/grib2/ACCESS-G3/IDY25026.APS3.wind.10m.2019092412.006.surface.grb2.

I have also tested it with a pyproj version 2.4.2.post1 and the results aren't much different. Here is the traceback from version 2.4.2.post1

> Traceback (most recent call last):
> File "", line 1, in
> File "/home/gar305/Documents/geostack/miniconda3/lib/python3.7/site-packages/pyproj/proj.py", line 146, in init
> self.crs = CRS.from_user_input(projparams if projparams is not None else kwargs)
> File "/home/gar305/Documents/geostack/miniconda3/lib/python3.7/site-packages/pyproj/crs.py", line 456, in from_user_input
> return cls(value)
> File "/home/gar305/Documents/geostack/miniconda3/lib/python3.7/site-packages/pyproj/crs.py", line 312, in init
> super(CRS, self).init(projstring)
> File "pyproj/_crs.pyx", line 1738, in pyproj._crs._CRS.init
> pyproj.exceptions.CRSError: Invalid projection: +proj=cyl +a=6371229 +b=6371229 +type=crs: (Internal Proj Error: proj_create: Error -5: unknown projection id)

I had initially opened this issue on [pyproj ](https://github.com/pyproj4/pyproj/issues/759#issue-761855882), but it seems that this is probably related to how pygrib assigns [projparams](https://github.com/jswhit/pygrib/blob/faafe6be8b67aa322cf47defc36d642f44b10dc7/pygrib.pyx#L1291), where for "regular_ll", it assigns "cyl". When I look at pyproj, I notice that for EPSG:4326, it requires "longlat". I am guessing that this is not an issue with Proj.4 but with how pygrib is assigning **projparams['proj']**.

**Comment by jswhit** [comment]
The projparams dict was not meant to be used to directly create a pyproj instance, although I suppose that would be useful.  I'll put that on to the todo list, but a pull request would be welcome.

**Comment by jswhit** [comment]
Shouldn't try to work on issues at 5pm on Friday.  Yes, the projparams dict should be able to be used to instantiate a pyproj instance, so this is  a bug.  The question is whether the 'regular_ll' grid type should have a projparams dict at all, since it is not really a projection.  Your suggestion of replacing 'cyl' with 'longlat' is a good one since is basically tells proj that there is no projection.  I will create a PR.

**Comment by jswhit** [comment]
FIxed in version 2.1.2
